pkg/operator/controller/dns/controller.go

```diff
@@ -661,26 +660,30 @@ func getIbmDNSProvider(dnsConfig *configv1.DNS, creds *corev1.Secret, instanceID
 	}
 
 	providerCfg := ibm.Config{
-		APIKey:     string(creds.Data["ibmcloud_api_key"]),
-		InstanceID: instanceID,
-		Zones:      zones,
-		UserAgent:  userAgent,
+		APIKey:    string(creds.Data["ibmcloud_api_key"]),
+		Zones:     zones,
+		UserAgent: userAgent,
 	}
 
 	if isPublic {
+		providerCfg.InstanceID = instanceCRN
 		provider, err := ibmpublicdns.NewProvider(providerCfg)
 		if err != nil {
 			return nil, fmt.Errorf("failed to initialize IBM CIS DNS provider: %w", err)
 		}
 		log.Info("successfully initialized IBM CIS DNS provider")
 		return provider, nil
 	} else {
+		matches := ibm.IBMResourceCRNRegexp.FindStringSubmatch(instanceCRN)
+		if len(matches) <= 0 {
+			return nil, fmt.Errorf("CRN does not match expected format: %s", instanceCRN)
```

Comment:
can we include that regex format we are expecting here otherwise this error is going to be really difficult to digest.

Reply:
Updated.